lib/class-wp-theme-json-gutenberg.php

```diff
@@ -835,6 +837,10 @@ private function get_block_styles( $style_nodes, $setting_nodes ) {
 			$selector     = $metadata['selector'];
 			$declarations = self::compute_style_properties( $node );
 			$block_rules .= self::to_ruleset( $selector, $declarations );
+
+			if ( self::ROOT_BLOCK_SELECTOR === $selector ) {
+				$block_rules .= ' .wp-site-blocks > * + *, body > * + * { margin-top: var( --wp-theme-block-gap ); margin-bottom: 0; }';
```

Comment:
This is introducing a small bug in the frontend (extra margin at the top of block templates), I'll see how to fix it properly.